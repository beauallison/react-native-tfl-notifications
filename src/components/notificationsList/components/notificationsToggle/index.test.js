import React from 'react';
import { shallow } from 'enzyme';
import Toggle from './index';

describe('components/notificationsList/notificationsToggle', () => {
  const props = {
    station: 'London',
    onValueChange: value => Promise.resolve(!value),
  };

  it('should render notifications toggle', () =>
    expect(shallow(<Toggle {...props} />))
      .toMatchSnapshot());

  it('should toggle switch', async () => {
    const wrapper = shallow(<Toggle {...props} />);
    wrapper.find('Switch').at(0).simulate('valueChange');

    setImmediate(() => expect(wrapper.state('enabled')).toBeTruthy());
  });

  it('should handle failed updating of notifications', async () => {
    const errorProps = { ...props };
    errorProps.onValueChange = () => Promise.reject(new Error('Error!'));

    const wrapper = shallow(<Toggle {...errorProps} />);
    wrapper.find('Switch').at(0).simulate('valueChange');

    setImmediate(() => expect(wrapper.state('enabled')).toBeFalsy());
  });
});
