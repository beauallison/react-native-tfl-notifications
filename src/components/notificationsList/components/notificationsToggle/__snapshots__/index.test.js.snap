// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/notificationsList/notificationsToggle should render notifications toggle 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NotificationsToggle
    enabled={false}
    onValueChange={[Function]}
    station="London"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
>
          London
</Text>,
        <Switch
          disabled={false}
          onValueChange={[Function]}
          style={
                    Object {
                              "marginBottom": 10,
                            }
          }
          value={false}
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": "London",
          "disabled": false,
          "ellipsizeMode": "tail",
        },
        "ref": null,
        "rendered": "London",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "disabled": false,
          "onValueChange": [Function],
          "style": Object {
            "marginBottom": 10,
          },
          "value": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
>
            London
</Text>,
          <Switch
            disabled={false}
            onValueChange={[Function]}
            style={
                        Object {
                                    "marginBottom": 10,
                                  }
            }
            value={false}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "London",
            "disabled": false,
            "ellipsizeMode": "tail",
          },
          "ref": null,
          "rendered": "London",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "disabled": false,
            "onValueChange": [Function],
            "style": Object {
              "marginBottom": 10,
            },
            "value": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
